### ISSUE TITLE
Compatibility with Mage 2.2.7 and 2.3 / PHP 7.2

### ISSUE THREAD

**Issue by simonmaass** [netresearch/dhl-module-shipping-m2#44]
**Feature description**
Any feedback for 2.2.7 and 2.3 support?

**Comment by ngolatka** [comment]
@simonmaass 
Can you be a bit more specific please?
* You asked for feedback. Was this an open issue? I don't see one.
* Did you try it and didn't it work?
* If it didn't work, what was the exact problem / error message / etc.?

**Comment by simonmaass** [comment]
@ngolatka we wanted to know if you have any experience with the compaibility of the 0.9.1 version with magento 2.3

we upgrade to 2.2.7 and it seems to work btw...

**Comment by simonmaass** [comment]
@ngolatka it is not an open issue but a question - when will there be an official 2.3 support?

**Comment by ngolatka** [comment]
Hi @simonmaass , thanks for clarifying. At the moment, there is no definitive answer to that question though.

Generally speaking, the development work will of course be based on current Magento versions, so there will _probably_ be support for Mage 2.3 in the future, but I can't tell you when that will happen.

**Comment by simonmaass** [comment]
are there any other ways in connecting your magento 2.3 with DHL? How are DHL customers suppose to work with magento? Would be great if support would be there soon!

**Comment by ngolatka** [comment]
We are currently checking what needs to be done in the DHL extension.

There is no timeline yet, but we will keep you posted in this issue.

**Comment by ngolatka** [comment]
The upcoming version 0.10.0 is going to be compatible with Magento 2.3.

Timeline: If everything goes according to plan and we don't get any surprises, the release should happen this month. Keep an eye on the [list of releases](https://github.com/netresearch/dhl-module-shipping-m2/releases).

**Comment by simonmaass** [comment]
@ngolatka thank you for the info!

**Comment by Mandrino** [comment]
This timely 2.3 support is greatly appreciated! Thanks!

**Comment by gewaechshaus** [comment]
Hi @ngolatka - please also add php 7.2 support. Many thanks

**Comment by ngolatka** [comment]
@gewaechshaus Already done. Version 0.10.0 will include support for PHP 7.2

**Comment by simonmaass** [comment]
is there any timeline that can be communicated? are u currently working on it? is it this week or next?

**Comment by ngolatka** [comment]
@simonmaass As I wrote above:
> the release should happen this month.

Yes, we are working on it right now, but we can't be more specific about the date. There's always the remote possibility that some unexpected last-minute changes need to be done.

**Comment by simonmaass** [comment]
any update here?

**Comment by ngolatka** [comment]
@simonmaass 
I have a keen eye on this. But please refrain from "ping" messages. I wrote:

> the release should happen this month.

This month is not over yet.

**Comment by ngolatka** [comment]
**A quick heads up:**
Earlier I wrote "_if we don't get any surprises, the release should happen this month_". I think I have tempted fate by saying that ;-)

The bloody flu knocked out our team. The remaining people are still working on the release, but it may take a bit longer (can't say how long exactly).

**Comment by simonmaass** [comment]
@ngolatka thnx for the heads up

**Comment by simonmaass** [comment]
any update here?

**Comment by Mandrino** [comment]
What seems to be happening is, that the relase of a ready-to-use update is blocked by "change requests" from DHL, as ngolatka said in another issue thread.

This is a procedure that I can't understand. Why are change requests withholding the relase of a new version? Why not release the ready thing and make the changes to the next (minor) update following shortly after?

These last-minute change requests have already been a problem for the last couple of years as they normally result in additional programming work, followed by additional testing work, followed by additional other problems.
Nobody seems to be thinking about eliminating the impact of such a "change request" on the update schedule.

It would be really helpful if there would be more frequent releases of more granular updates, like compatibility, so we can work on our servers instead of waiting for feature or behavior change requests by DHL impacting the relase of compatibility updates.

Please think about making this more granular. Magento 2.3 is now more than 3 months old and will likely soon be replaced by Magento 2.4 - while we still don't have compatibility with the old 2.3.

Thanks

**Comment by Fgruntjes** [comment]
This is also for us a serious problem. Especially for new projects it is just not really acceptable to release a new project on an older version of Magento. In this case also PHP 7.2 for the extra performance gain. PHP 7.2 has been released over 3 months now. Making the compatibility changes is a small effort.

If the compatibility issues are not solved faster we will in the feature advise customers against DHL because of this.

**Comment by diacu** [comment]
I think not having a compatible version with 2.3 is a serious problem for many of us. Can you guys at least try to release a beta version, with none of the change requests included?

**Comment by ngolatka** [comment]
Please calm down everyone. We are fully aware of the importance of this and are doing our best not to leave anybody hanging.

We are pushing to get the new module version out ASAP.

At the same time, we are discussing possible options for more frequent / more granular releases.

**Comment by Fgruntjes** [comment]
@ngolatka thank you for your response.

What is the expected time schedule for the next version with at least 2.3 / php72 support?

**Comment by ngolatka** [comment]
We received the final "Go" yesterday from DHL.

Next step is the technical review from Magento Marketplace. This is needed because we don't want to push something to Github which would fail Magento's guidelines.

If the technical review gives us a green light, we will push the release to Github, and at the same time start the full release process on Magento Marketplace.

----

Some background info:

**Technical reviews by Magento** are an automated process which happens quite fast (typically much less than a day). If that review necessitates changes on our side, we need of course time to implement those.

A **full Marketplace release**, on the other hand, can (and usually does) take considerably longer, but that's not directly relevant because we can push to Github in parallel.

You might notice the lack of any fixed dates in my above reply. That's because **we simply do not know**. We are dependent on cooperation from third parties - now Magento Marketplace, to be precise.

The DHL module is not "our own", we are developing and maintaining it _for DHL_.

**Comment by ngolatka** [comment]
**The new DHL Shipping version 0.10.0 is now available:**

https://github.com/netresearch/dhl-module-shipping-m2/releases/tag/0.10.0

**Comment by jotex-jordan** [comment]
Can this module work with DHL Express because we have tried it with DHL express account in Jordan and it did work and print label

**Comment by ngolatka** [comment]
@jotex-jordan The DHL Shipping module for Magento 2 supports only a limited set of shipping origins (sender country). Unfortunately, Jordan is not in [the list](https://marketplace.magento.com/media/catalog/product/dhl-module-shipping-m2-0-9-0-ce/user_guides.pdf) (page 4).
So the answer to your question is: sorry, no, this module is not the right one for you.

**Comment by jotex-jordan** [comment]
I am not a developer but a merchant
I have tested the DHL default for magneto 2.3 and I can get tracking number and print label
Of course our account is DHL express
So how come its working and please

**Comment by jotex-jordan** [comment]
Dear

I am not a developer but a merchant
I have tested the DHL default for magneto 2.3 and I can get tracking number and print label
Of course our account is DHL express
So how come its working and please

Mike

------------------------------------
Best Regards,

Shireen General Trading

Build. 4   Al-Mutanabbi St.
Jabal Amman, Third Circle
Behind La-Royal Hotel
Amman, Jordan 11183

Tel:  +962 6 4650025
Fax: +962 6 4650026
Mobile: +962 79 627 3377
Skype: shireentrading

www.shireentrading.com<http://www.shireentrading.com/>
www.jotex.jo<http://www.jotex.jo/>

**Comment by ngolatka** [comment]
@jotex-jordan I'm sorry, but your question is not related to the DHL Shipping module, therefore you need to ask it elsewhere, not in this Github project.

This project is not for support questions about the DHL default integration in Magento, and also not for questions about DHL Express. Both have nothing to do with [this module](https://github.com/netresearch/dhl-module-shipping-m2).
